example.py

```diff
@@ -44,63 +54,60 @@ def heightAt(x, z):
     # Heightmap coordinates are not equal to world coordinates!
     return heightmap[(x - area[0], z - area[1])]
 
-def buildHouse(x1, y1, z1, x2, y2, z2, interface):
+
+def buildHouse(x1, y1, z1, x2, y2, z2):
     """Build a small house."""
     # floor
-    for x in range(x1, x2):
-        for z in range(z1, z2):
-            interface.setBlock(x, y1, z, "cobblestone")
+    interface.fill(x1, y1, z1, x2 - 1, y1, z2 - 1, "cobblestone")
+
     # walls
-    for y in range(y1 + 1, y2):
-        for x in range(x1 + 1, x2 - 1):
-            interface.setBlock(x, y, z1, "oak_planks")
-            interface.setBlock(x, y, z2 - 1, "oak_planks")
-        for z in range(z1 + 1, z2 - 1):
-            interface.setBlock(x1, y, z, "oak_planks")
-            interface.setBlock(x2 - 1, y, z, "oak_planks")
+    interface.fill(x1 + 1, y1, z1, x2 - 2, y2, z1, "oak_planks")
+    interface.fill(x1 + 1, y1, z2 - 1, x2 - 2, y2, z2 - 1, "oak_planks")
+    interface.fill(x1, y1, z1 + 1, x1, y2, z2 - 2, "oak_planks")
+    interface.fill(x2 - 1, y1, z1 + 1, x2 - 1, y2, z2 - 2, "oak_planks")
+
     # corners
-    for dx in range(2):
-        for dz in range(2):
-            x = x1 + dx * (x2 - x1 - 1)
-            z = z1 + dz * (z2 - z1 - 1)
-            for y in range(y1, y2):
-                interface.setBlock(x, y, z, "oak_log")
+    interface.fill(x1, y1, z1, x1, y2, z1, "oak_log")
+    interface.fill(x2 - 1, y1, z1, x2 - 1, y2, z1, "oak_log")
+    interface.fill(x1, y1, z2 - 1, x1, y2, z2 - 1, "oak_log")
+    interface.fill(x2 - 1, y1, z2 - 1, x2 - 1, y2, z2 - 1, "oak_log")
+
     # clear interior
     for y in range(y1 + 1, y2):
         for x in range(x1 + 1, x2 - 1):
             for z in range(z1 + 1, z2 - 1):
-                interface.setBlock(x, y, z, "air")
+                # check what's at that place and only delete if not air
+                if "air" not in interface.getBlock(x, y, z):
```

Comment:
I was thinking to use it as an example for getBlock and setBlock since they don't turn up in the rest of the code, but I don't mind reverting it.

Reply:
Ah ok, makes sense. Hm I don't think this is the way to do it, since I don't think it's a good example for using getBlock, but it would be cool if there was something that uses getBlock in the example.

Reply:
Agreed. I'll leave it like that but we can rework it when reworking the example/s.